internal/internal_task_pollers.go

```diff
@@ -1008,7 +1019,14 @@ func (atp *activityTaskPoller) poll(ctx context.Context) (*s.PollForActivityTask
 	if err != nil {
 		retryable := isServiceTransientError(err)
 		if retryable {
-			atp.metricsScope.Counter(metrics.ActivityPollTransientFailedCounter).Inc(1)
+
+			if target := (*s.ServiceBusyError)(nil); errors.As(err, &target) {
```

Comment:
I kinda wonder if it'd be worth making a helper specifically for service-busy, as it's handled specially in a few locations...

but not at all requiring that now, just thinking out loud.